From 317ab997bf9870b2d90fb92db665a799d9e03120 Mon Sep 17 00:00:00 2001
From: zack <zack@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Mon, 29 May 2000 19:33:52 +0000
Subject: [PATCH] 	* hashtab.c, partition.c, sort.c, xmemdup.c: Include
 string.h 	if HAVE_STRING_H. 	* pexecute.c,
 xexit.c: Include stdlib.h if HAVE_STDLIB_H. 	*
 objalloc.c: Include config.h.  Include stdlib.h and don't 
 declare malloc or free if HAVE_STDLIB_H. 	*
 strerror.c, strsignal.c: Include stdlib.h if HAVE_STDLIB_H,
 	else declare malloc without prototype.  Include
 string.h if 	HAVE_STRING_H, else declare memset without
 prototype.  Don't 	include stddef.h.

git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/trunk@34254 138bc75d-0d04-0410-961f-82ee72b054a4
---
 libiberty/ChangeLog   | 12 ++++++++++++
 libiberty/hashtab.c   |  4 ++++
 libiberty/objalloc.c  |  7 +++++++
 libiberty/partition.c |  4 ++++
 libiberty/pexecute.c  |  3 +++
 libiberty/sort.c      |  3 +++
 libiberty/strerror.c  | 19 +++++++++++--------
 libiberty/strsignal.c | 19 +++++++++++--------
 libiberty/xexit.c     |  3 +++
 libiberty/xmemdup.c   |  3 +++
 10 files changed, 61 insertions(+), 16 deletions(-)

2000-05-29  Zack Weinberg  <zack@wolery.cumb.org>

	* hashtab.c, partition.c, sort.c, xmemdup.c: Include string.h
	if HAVE_STRING_H.
	* pexecute.c, xexit.c: Include stdlib.h if HAVE_STDLIB_H.
	* objalloc.c: Include config.h.  Include stdlib.h and don't
	declare malloc or free if HAVE_STDLIB_H.
	* strerror.c, strsignal.c: Include stdlib.h if HAVE_STDLIB_H,
	else declare malloc without prototype.  Include string.h if
	HAVE_STRING_H, else declare memset without prototype.  Don't
	include stddef.h.

diff --git a/libiberty/objalloc.c b/libiberty/objalloc.c
index 57754a8..5099569 100644
--- a/libiberty/objalloc.c
+++ b/libiberty/objalloc.c
@@ -18,6 +18,8 @@ Foundation, 59 Temple Place - Suite 330,
 Boston, MA 02111-1307, USA.  */
 
 #include "ansidecl.h"
+#include "config.h"
+
 #include "objalloc.h"
 
 /* Get a definition for NULL.  */
@@ -33,11 +35,16 @@ Boston, MA 02111-1307, USA.  */
 #include <stddef.h>
 #endif
 
+#ifdef HAVE_STDLIB_H
+#include <stdlib.h>
+#else
 /* For systems with larger pointers than ints, this must be declared.  */
 extern PTR malloc PARAMS ((size_t));
 extern void free PARAMS ((PTR));
 #endif
 
+#endif
+
 /* These routines allocate space for an object.  Freeing allocated
    space may or may not free all more recently allocated space.
 
diff --git a/libiberty/pexecute.c b/libiberty/pexecute.c
index 56ddec7..f1aea9d 100644
--- a/libiberty/pexecute.c
+++ b/libiberty/pexecute.c
@@ -35,6 +35,9 @@ Boston, MA 02111-1307, USA.  */
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
 #endif
+#ifdef HAVE_STDLIB_H
+#include <stdlib.h>
+#endif
 #define ISSPACE (x) isspace(x)
 #ifdef HAVE_SYS_WAIT_H
 #include <sys/wait.h>
diff --git a/libiberty/strerror.c b/libiberty/strerror.c
index 644cc75..0dd2220 100644
--- a/libiberty/strerror.c
+++ b/libiberty/strerror.c
@@ -25,14 +25,17 @@
 
 /*  Routines imported from standard C runtime libraries. */
 
-#ifdef __STDC__
-#include <stddef.h>
-extern void *malloc (size_t size);				/* 4.10.3.3 */
-extern void *memset (void *s, int c, size_t n);			/* 4.11.6.1 */
-#else	/* !__STDC__ */
-extern char *malloc ();		/* Standard memory allocater */
-extern char *memset ();
-#endif	/* __STDC__ */
+#ifdef HAVE_STDLIB_H
+#include <stdlib.h>
+#else
+extern PTR malloc ();
+#endif
+
+#ifdef HAVE_STRING_H
+#include <string.h>
+#else
+extern PTR memset ();
+#endif
 
 #ifndef MAX
 #  define MAX(a,b) ((a) > (b) ? (a) : (b))
diff --git a/libiberty/strsignal.c b/libiberty/strsignal.c
index c7bb10c..5625323 100644
--- a/libiberty/strsignal.c
+++ b/libiberty/strsignal.c
@@ -23,14 +23,17 @@
 
 /*  Routines imported from standard C runtime libraries. */
 
-#ifdef __STDC__
-#include <stddef.h>
-extern void *malloc (size_t size);				/* 4.10.3.3 */
-extern void *memset (void *s, int c, size_t n);			/* 4.11.6.1 */
-#else	/* !__STDC__ */
-extern char *malloc ();		/* Standard memory allocater */
-extern char *memset ();
-#endif	/* __STDC__ */
+#ifdef HAVE_STDLIB_H
+#include <stdlib.h>
+#else
+extern PTR malloc ();
+#endif
+
+#ifdef HAVE_STRING_H
+#include <string.h>
+#else
+extern PTR memset ();
+#endif
 
 /* Undefine the macro we used to hide the definition of sys_siglist
    found in the system header files.  */
diff --git a/libiberty/xexit.c b/libiberty/xexit.c
index 431bbe0..7aa52ac 100644
--- a/libiberty/xexit.c
+++ b/libiberty/xexit.c
@@ -21,6 +21,9 @@ Boston, MA 02111-1307, USA.  */
 #include "libiberty.h"
 
 #include <stdio.h>
+#ifdef HAVE_STDLIB_H
+#include <stdlib.h>
+#endif
 
 /* This variable is set by xatexit if it is called.  This way, xmalloc
    doesn't drag xatexit into the link.  */
-- 
1.8.0

